paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -396,6 +440,88 @@ def sync_secrets(
     return True
 
 
+def sync_datastore_credentials(
+    kube_client: KubeClient,
+    cluster: str,
+    service: str,
+    secret_provider_name: str,
+    vault_cluster_config: Dict[str, str],
+    soa_dir: str,
+    vault_token_file: str,
+) -> bool:
```

Comment:
Apologies if I've missed this, but our other `sync_X` functions take in an optional `namespace` since the `paasta_secrets_sync` script can take it in as an argument ( eg https://github.com/Yelp/paasta/blob/master/paasta_tools/kubernetes/bin/paasta_secrets_sync.py#L335 + https://github.com/Yelp/paasta/blob/master/paasta_tools/kubernetes/bin/paasta_secrets_sync.py#L393).

Is there a specific reason database credentials won't need this/will _only_ ever sync to the service instance's namespace? 

That may become an issue when using db creds in e.g. tron jobs or spark jobs?

Reply:
we eventually call `namespace=instance_config.get_namespace()` in the `create_or_update_k8s_secret(` call, which should reference the same I think? looks like sync_boto and sync_crypto do the same

Reply:
oh, good catch!

Reply:
Unless we're overriding the `instance_config`'s namespace with the one provided as an argument to this script, I suspect that boto & crypto secrets are also never syncing to the requested namespace if one is provided (entirely possible, I think there is more work to be done to support boto_keys in tron for example, so we wouldn't have discovered this yet!).  We also pass a namespace rather than use the instance config's for spark secrets.

In retrospect, I do know that we currently have someone working on a known issue with https://github.com/Yelp/paasta/blob/master/paasta_tools/kubernetes/bin/paasta_secrets_sync.py#L147C5-L147C48 in that it also does not honor the passed namespace, so for now, I think `instance_config.get_namespace()` is fine and should cover the default usecase, just like with boto/crypto.  You can follow https://jira.yelpcorp.com/browse/COMPINFRA-2860 for updates to that issue.

Just something to be aware about that we may be changing this soon and be warned if any other workloads in non-standard namespaces start to need these secrets!

Reply:
ah I see, I can imagine this affecting us as well for non-default-namespaced paasta services.

let me know if I should make that change as part of this PR or leave that to the COMPINFRA ticket's refactor

Reply:
I think one quick improvement that will help us prep for the future and will not have to change too much in the future is to have your `sync_datastore_credentials` accept a `namespace` and default it to the `override_namespace` arg provided in the `sync_all_secrets` function.  If it is none/empty, use `instance_config.get_namespace()` just as you do already.

Reply:
the var seems to be named `overwrite_namespace` - might be a phonetic typo of override? 😄

Reply:
Yup, apologies, I meant `overwrite_namespace`! this one https://github.com/Yelp/paasta/blob/master/paasta_tools/kubernetes/bin/paasta_secrets_sync.py#L263

Reply:
done! thanks for pointing this out